service/worker/scheduler/workflow.go

```diff
@@ -523,6 +526,7 @@ func (s *scheduler) getFutureActionTimes(n int) []*time.Time {
 }
 
 func (s *scheduler) handleDescribeQuery() (*schedspb.DescribeResponse, error) {
+	// this field is never read by the workflow so it's okay to modify it from a query
 	s.Info.FutureActionTimes = s.getFutureActionTimes(s.tweakables.FutureActionCount)
```

Comment:
if that is the case, we should create a local variable in this query handler for this. Modify state value in query is a problem waiting to be triggered.